sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -176,9 +192,16 @@ object NestedColumnAliasing {
         // By default, ColumnPruning rule uses `attr` already.
         if (nestedFieldToAlias.nonEmpty &&
             nestedFieldToAlias
-              .map { case (nestedField, _) => totalFieldNum(nestedField.dataType) }
-              .sum < totalFieldNum(attr.dataType)) {
-          Some(attr.exprId -> nestedFieldToAlias)
+              .foldLeft(Seq[ExtractValue]()) {
+                (unique, curr) => if (!unique.exists(curr._1.semanticEquals(_))) {
+                  curr._1 +: unique
+                } else {
+                  unique
+                }
+              }
+              .map { t => totalFieldNum(t.dataType)  }
+              .sum < totalFieldNum(attr._2)) {
```

Comment:
Its hard to read this part. Could you pull the left value out from the `if` condition?
```
val xxx = yyy
if (xxx < totalFieldNum(attr._2)) { ...
```

Reply:
Sure.

Reply:
Updated.